<!-- Generated by pkgdown: do not edit by hand -->
<!DOCTYPE html>
<html lang="en">
  <head>
  <meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">

<title>An example FunctionalSingleCellExperiment — fsce_small • scrunchy</title>

<!-- favicons -->
<link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png">
<link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png">
<link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png" />
<link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png" />
<link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png" />
<link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png" />
<!-- jquery -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.3.1/jquery.min.js" integrity="sha256-FgpCb/KJQlLNfOu91ta32o/NMZxltwRo8QtmkMRdAu8=" crossorigin="anonymous"></script>
<!-- Bootstrap -->
<link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.3.7/sandstone/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous" />

<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.3.7/js/bootstrap.min.js" integrity="sha256-U5ZEeKfGNOja007MMD3YBI0A3OSZOQbeG6z2f2Y0hu8=" crossorigin="anonymous"></script>

<!-- Font Awesome icons -->
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/4.7.0/css/font-awesome.min.css" integrity="sha256-eZrrJcwDc/3uDhsdt61sL2oOBY362qM3lon1gyExkL0=" crossorigin="anonymous" />

<!-- clipboard.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.4/clipboard.min.js" integrity="sha256-FiZwavyI2V6+EXO1U+xzLG3IKldpiTFf3153ea9zikQ=" crossorigin="anonymous"></script>

<!-- sticky kit -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/sticky-kit/1.1.3/sticky-kit.min.js" integrity="sha256-c4Rlo1ZozqTPE2RLuvbusY3+SU1pQaJC0TjuhygMipw=" crossorigin="anonymous"></script>

<!-- pkgdown -->
<link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script>



<meta property="og:title" content="An example FunctionalSingleCellExperiment — fsce_small" />

<meta property="og:description" content="Contains sample data for a functional single-cell experiment." />

<meta property="og:image" content="https://scrunchy.hesselberthlab.org/logo.png" />
<meta name="twitter:card" content="summary" />



<!-- mathjax -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script>

<!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->


  </head>

  <body>
    <div class="container template-reference-topic">
      <header>
      <div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">scrunchy</a>
        <span class="version label label-danger" data-toggle="tooltip" data-placement="bottom" title="Unreleased version">0.0.0.9000</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="../index.html">
    <span class="fa fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../articles/scrunchy.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul>
      
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/hesselberthlab/scrunchy">
    <span class="fa fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
      
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      
      </header>

<div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>An example FunctionalSingleCellExperiment</h1>
    <small class="dont-index">Source: <a href='https://github.com/hesselberthlab/scrunchy/blob/master/R/data.R'><code>R/data.R</code></a></small>
    <div class="hidden name"><code>fsce_small.Rd</code></div>
    </div>

    <div class="ref-description">
    
    <p>Contains sample data for a functional single-cell experiment.</p>
    
    </div>

    <pre class="usage"><span class='no'>fsce_small</span></pre>
        
    <h2 class="hasAnchor" id="format"><a class="anchor" href="#format"></a>Format</h2>

    <p>An object of class <code>FunctionalSingleCellExperiment</code> of length 2.</p>
    
    <h2 class="hasAnchor" id="source"><a class="anchor" href="#source"></a>Source</h2>

    <p>Generated by <code>data-raw/fsce_small.R</code>.</p>
    
    <h2 class="hasAnchor" id="details"><a class="anchor" href="#details"></a>Details</h2>

    <p>There are two <code>SingleCellExperiments</code>:</p><ol>
<li><p><code>rnaseq</code>: a <code>SingleCellExperiment</code> object containing a single-cell mRNA sequencing
(10x Genomics V2 3 prime) cell/gene counts matrix in the <code>counts</code> slot.
Log-normalized counts are in the <code>logcounts</code> slot.</p></li>
<li><p><code>haircut</code>: a <code>SingleCellExperiment</code> object containing a Haircut experiment using DNA
repair oligos. Counts are in the <code>counts</code> slot, and centered log-ratio
normalized counts are in the <code>logcounts</code> slot.</p></li>
</ol>
    <p>Haircut repair substrates include:</p><ul>
<li><p>A "normal" DNA oligo (<code>Normal_</code>)</p></li>
<li><p>a ribonucleotide (rG) at position 45 (<code>Ribo_</code>)</p></li>
<li><p>an Abasic site at position 45 (<code>Abasic_</code>)</p></li>
<li><p>a uracil:adenosine base-pair, with uracil at position 45 (<code>Uracil_</code>)</p></li>
<li><p>a uracil:guanosine base-pair, with uracil at position 45 (<code>GU_</code>)</p></li>
</ul>
    <p>Dimensionality reduction calculations are in the <code>rnaseq</code> experiment, including
<code>PCA</code>, <code>UMAP</code>, and <code>TSNE</code> slots. In addition, k-means clusters are available as
<code>rowData</code>.</p>
    

    <h2 class="hasAnchor" id="examples"><a class="anchor" href="#examples"></a>Examples</h2>
    <pre class="examples"><div class='input'><span class='no'>fsce_small</span></div><div class='output co'>#&gt; A FunctionalSingleCellExperiment object of 2 listed
#&gt;  experiments with user-defined names and respective classes. 
#&gt;  Containing an ExperimentList class object of length 2: 
#&gt;  [1] rnaseq: SingleCellExperiment with 9479 rows and 250 columns 
#&gt;  [2] haircut: SingleCellExperiment with 426 rows and 250 columns 
#&gt; Features: 
#&gt;  experiments() - obtain the ExperimentList instance 
#&gt;  colData() - the primary/phenotype DataFrame 
#&gt;  sampleMap() - the sample availability DataFrame 
#&gt;  `$`, `[`, `[[` - extract colData columns, subset, or experiment 
#&gt;  *Format() - convert into a long or wide DataFrame 
#&gt;  assays() - convert ExperimentList to a SimpleList of matrices</div><div class='input'>
<span class='co'># Individual experiments</span>
<span class='no'>fsce_small</span><span class='kw'>[[</span><span class='st'>"rnaseq"</span>]]</div><div class='output co'>#&gt; class: SingleCellExperiment 
#&gt; dim: 9479 250 
#&gt; metadata(0):
#&gt; assays(2): counts logcounts
#&gt; rownames(9479): FO538757.2 AP006222.2 ... AC004556.1 AC240274.1
#&gt; rowData names(0):
#&gt; colnames(250): TGCGGGTTCAACTCTT TGGTTCCTCCCAAGAT ... CTCGAAACACACTGCG
#&gt;   TGACGGCAGAGTCGGT
#&gt; colData names(2): cell_id k_cluster
#&gt; reducedDimNames(3): PCA UMAP TSNE
#&gt; spikeNames(0):</div><div class='input'>
<span class='no'>fsce_small</span><span class='kw'>[[</span><span class='st'>"haircut"</span>]]</div><div class='output co'>#&gt; class: SingleCellExperiment 
#&gt; dim: 426 250 
#&gt; metadata(0):
#&gt; assays(2): counts logcounts
#&gt; rownames(426): Abasic_1 Abasic_10 ... riboG_8 riboG_9
#&gt; rowData names(2): hairpin position
#&gt; colnames(250): TGCGGGTTCAACTCTT TGGTTCCTCCCAAGAT ... CTCGAAACACACTGCG
#&gt;   TGACGGCAGAGTCGGT
#&gt; colData names(1): cell_id
#&gt; reducedDimNames(0):
#&gt; spikeNames(0):</div><div class='input'>
<span class='co'># Gene and activity names (first 5 items)</span>
<span class='fu'><a href='https://www.rdocumentation.org/packages/base/topics/colnames'>rownames</a></span>(<span class='no'>fsce_small</span><span class='kw'>[[</span><span class='st'>"rnaseq"</span>]])[<span class='fl'>1</span>:<span class='fl'>5</span>]</div><div class='output co'>#&gt; [1] "FO538757.2" "AP006222.2" "NOC2L"      "ISG15"      "TNFRSF18"  </div><div class='input'>
<span class='fu'><a href='https://www.rdocumentation.org/packages/base/topics/colnames'>rownames</a></span>(<span class='no'>fsce_small</span><span class='kw'>[[</span><span class='st'>"haircut"</span>]])[<span class='fl'>1</span>:<span class='fl'>5</span>]</div><div class='output co'>#&gt; [1] "Abasic_1"  "Abasic_10" "Abasic_11" "Abasic_12" "Abasic_13"</div><div class='input'>
<span class='co'># subset with `[` using `[features, cell_ids, experiments]`</span>
<span class='no'>features</span> <span class='kw'>&lt;-</span> <span class='fu'><a href='https://www.rdocumentation.org/packages/base/topics/c'>c</a></span>(<span class='st'>"Uracil_45"</span>, <span class='st'>"TP53"</span>)
<span class='no'>cell_ids</span> <span class='kw'>&lt;-</span> <span class='fu'><a href='https://www.rdocumentation.org/packages/base/topics/c'>c</a></span>(<span class='st'>"GAAACTCTCCGAAGAG"</span>, <span class='st'>"CGCTATCGTTCACCTC"</span>)

<span class='no'>fsce_small</span>[<span class='no'>features</span>, <span class='no'>cell_ids</span>, ]</div><div class='output co'>#&gt; <span class='warning'>Warning: Not all identifiers found in data</span></div><div class='output co'>#&gt; A FunctionalSingleCellExperiment object of 2 listed
#&gt;  experiments with user-defined names and respective classes. 
#&gt;  Containing an ExperimentList class object of length 2: 
#&gt;  [1] rnaseq: SingleCellExperiment with 1 rows and 1 columns 
#&gt;  [2] haircut: SingleCellExperiment with 1 rows and 1 columns 
#&gt; Features: 
#&gt;  experiments() - obtain the ExperimentList instance 
#&gt;  colData() - the primary/phenotype DataFrame 
#&gt;  sampleMap() - the sample availability DataFrame 
#&gt;  `$`, `[`, `[[` - extract colData columns, subset, or experiment 
#&gt;  *Format() - convert into a long or wide DataFrame 
#&gt;  assays() - convert ExperimentList to a SimpleList of matrices</div><div class='input'>
<span class='fu'><a href='https://www.rdocumentation.org/packages/base/topics/library'>library</a></span>(<span class='no'>SingleCellExperiment</span>)</div><div class='output co'>#&gt; <span class='message'>Loading required package: SummarizedExperiment</span></div><div class='output co'>#&gt; <span class='message'>Loading required package: GenomicRanges</span></div><div class='output co'>#&gt; <span class='message'>Loading required package: stats4</span></div><div class='output co'>#&gt; <span class='message'>Loading required package: BiocGenerics</span></div><div class='output co'>#&gt; <span class='message'>Loading required package: parallel</span></div><div class='output co'>#&gt; <span class='message'></span>
#&gt; <span class='message'>Attaching package: ‘BiocGenerics’</span></div><div class='output co'>#&gt; <span class='message'>The following objects are masked from ‘package:parallel’:</span>
#&gt; <span class='message'></span>
#&gt; <span class='message'>    clusterApply, clusterApplyLB, clusterCall, clusterEvalQ,</span>
#&gt; <span class='message'>    clusterExport, clusterMap, parApply, parCapply, parLapply,</span>
#&gt; <span class='message'>    parLapplyLB, parRapply, parSapply, parSapplyLB</span></div><div class='output co'>#&gt; <span class='message'>The following object is masked from ‘package:scrunchy’:</span>
#&gt; <span class='message'></span>
#&gt; <span class='message'>    normalize</span></div><div class='output co'>#&gt; <span class='message'>The following objects are masked from ‘package:stats’:</span>
#&gt; <span class='message'></span>
#&gt; <span class='message'>    IQR, mad, sd, var, xtabs</span></div><div class='output co'>#&gt; <span class='message'>The following objects are masked from ‘package:base’:</span>
#&gt; <span class='message'></span>
#&gt; <span class='message'>    anyDuplicated, append, as.data.frame, basename, cbind, colMeans,</span>
#&gt; <span class='message'>    colnames, colSums, dirname, do.call, duplicated, eval, evalq,</span>
#&gt; <span class='message'>    Filter, Find, get, grep, grepl, intersect, is.unsorted, lapply,</span>
#&gt; <span class='message'>    lengths, Map, mapply, match, mget, order, paste, pmax, pmax.int,</span>
#&gt; <span class='message'>    pmin, pmin.int, Position, rank, rbind, Reduce, rowMeans, rownames,</span>
#&gt; <span class='message'>    rowSums, sapply, setdiff, sort, table, tapply, union, unique,</span>
#&gt; <span class='message'>    unsplit, which, which.max, which.min</span></div><div class='output co'>#&gt; <span class='message'>Loading required package: S4Vectors</span></div><div class='output co'>#&gt; <span class='message'></span>
#&gt; <span class='message'>Attaching package: ‘S4Vectors’</span></div><div class='output co'>#&gt; <span class='message'>The following object is masked from ‘package:base’:</span>
#&gt; <span class='message'></span>
#&gt; <span class='message'>    expand.grid</span></div><div class='output co'>#&gt; <span class='message'>Loading required package: IRanges</span></div><div class='output co'>#&gt; <span class='message'>Loading required package: GenomeInfoDb</span></div><div class='output co'>#&gt; <span class='message'>Loading required package: Biobase</span></div><div class='output co'>#&gt; <span class='message'>Welcome to Bioconductor</span>
#&gt; <span class='message'></span>
#&gt; <span class='message'>    Vignettes contain introductory material; view with</span>
#&gt; <span class='message'>    'browseVignettes()'. To cite Bioconductor, see</span>
#&gt; <span class='message'>    'citation("Biobase")', and for packages 'citation("pkgname")'.</span></div><div class='output co'>#&gt; <span class='message'>Loading required package: DelayedArray</span></div><div class='output co'>#&gt; <span class='message'>Loading required package: matrixStats</span></div><div class='output co'>#&gt; <span class='message'></span>
#&gt; <span class='message'>Attaching package: ‘matrixStats’</span></div><div class='output co'>#&gt; <span class='message'>The following objects are masked from ‘package:Biobase’:</span>
#&gt; <span class='message'></span>
#&gt; <span class='message'>    anyMissing, rowMedians</span></div><div class='output co'>#&gt; <span class='message'>Loading required package: BiocParallel</span></div><div class='output co'>#&gt; <span class='message'></span>
#&gt; <span class='message'>Attaching package: ‘DelayedArray’</span></div><div class='output co'>#&gt; <span class='message'>The following objects are masked from ‘package:matrixStats’:</span>
#&gt; <span class='message'></span>
#&gt; <span class='message'>    colMaxs, colMins, colRanges, rowMaxs, rowMins, rowRanges</span></div><div class='output co'>#&gt; <span class='message'>The following objects are masked from ‘package:base’:</span>
#&gt; <span class='message'></span>
#&gt; <span class='message'>    aperm, apply</span></div><div class='input'><span class='co'># dimensionality reduction results</span>
<span class='fu'><a href='https://www.rdocumentation.org/packages/SingleCellExperiment/topics/SCE-reducedDims'>reducedDimNames</a></span>(<span class='no'>fsce_small</span><span class='kw'>[[</span><span class='st'>"rnaseq"</span>]])</div><div class='output co'>#&gt; [1] "PCA"  "UMAP" "TSNE"</div><div class='input'>
<span class='co'># UMAP results (first 5 rows)</span>
<span class='fu'><a href='https://www.rdocumentation.org/packages/SingleCellExperiment/topics/SCE-reducedDims'>reducedDim</a></span>(<span class='no'>fsce_small</span><span class='kw'>[[</span><span class='st'>"rnaseq"</span>]], <span class='st'>"UMAP"</span>)[<span class='fl'>1</span>:<span class='fl'>5</span>, ]</div><div class='output co'>#&gt;                       [,1]      [,2]
#&gt; TGCGGGTTCAACTCTT  2.146147 3.9898103
#&gt; TGGTTCCTCCCAAGAT  1.376538 3.6447333
#&gt; CATATTCCATGCCCGA  3.815146 2.7023185
#&gt; TCACAAGTCTCCAACC -0.216390 0.2069193
#&gt; GGACAAGAGCCCTAAT  2.566830 3.0588671</div><div class='input'>
<span class='co'># k-means cluster IDs</span>
<span class='fu'>colData</span>(<span class='no'>fsce_small</span><span class='kw'>[[</span><span class='st'>"rnaseq"</span>]])</div><div class='output co'>#&gt; DataFrame with 250 rows and 2 columns
#&gt;                           cell_id   k_cluster
#&gt;                       &lt;character&gt; &lt;character&gt;
#&gt; TGCGGGTTCAACTCTT TGCGGGTTCAACTCTT           6
#&gt; TGGTTCCTCCCAAGAT TGGTTCCTCCCAAGAT           6
#&gt; CATATTCCATGCCCGA CATATTCCATGCCCGA           6
#&gt; TCACAAGTCTCCAACC TCACAAGTCTCCAACC           6
#&gt; GGACAAGAGCCCTAAT GGACAAGAGCCCTAAT           6
#&gt; ...                           ...         ...
#&gt; CTACGTCCAGTGGAGT CTACGTCCAGTGGAGT           4
#&gt; TTAGGCACAGGTGGAT TTAGGCACAGGTGGAT           2
#&gt; AGCGTCGTCGTCGTTC AGCGTCGTCGTCGTTC           4
#&gt; CTCGAAACACACTGCG CTCGAAACACACTGCG           6
#&gt; TGACGGCAGAGTCGGT TGACGGCAGAGTCGGT           6</div><div class='input'>
</div></pre>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="sidebar">
    <h2>Contents</h2>
    <ul class="nav nav-pills nav-stacked">
      
      <li><a href="#format">Format</a></li>

      <li><a href="#source">Source</a></li>

      <li><a href="#details">Details</a></li>
      
      <li><a href="#examples">Examples</a></li>
    </ul>

  </div>
</div>

      <footer>
      <div class="copyright">
  <p>Developed by <a href='https://hesselberthlab.org'>Jay Hesselberth</a>, Amanda Richer, Kent Riemondy.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.3.0.</p>
</div>
      </footer>
   </div>

  

  </body>
</html>

