<!-- Generated by pkgdown: do not edit by hand -->
<!DOCTYPE html>
<html lang="en">
  <head>
  <meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">

<title>Function reference â€¢ scrunchy</title>

<!-- favicons -->
<link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png">
<link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png">
<link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png" />
<link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png" />
<link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png" />
<link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png" />
<!-- jquery -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.3.1/jquery.min.js" integrity="sha256-FgpCb/KJQlLNfOu91ta32o/NMZxltwRo8QtmkMRdAu8=" crossorigin="anonymous"></script>
<!-- Bootstrap -->
<link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.3.7/sandstone/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous" />

<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.3.7/js/bootstrap.min.js" integrity="sha256-U5ZEeKfGNOja007MMD3YBI0A3OSZOQbeG6z2f2Y0hu8=" crossorigin="anonymous"></script>

<!-- Font Awesome icons -->
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/4.7.0/css/font-awesome.min.css" integrity="sha256-eZrrJcwDc/3uDhsdt61sL2oOBY362qM3lon1gyExkL0=" crossorigin="anonymous" />

<!-- clipboard.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.4/clipboard.min.js" integrity="sha256-FiZwavyI2V6+EXO1U+xzLG3IKldpiTFf3153ea9zikQ=" crossorigin="anonymous"></script>

<!-- sticky kit -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/sticky-kit/1.1.3/sticky-kit.min.js" integrity="sha256-c4Rlo1ZozqTPE2RLuvbusY3+SU1pQaJC0TjuhygMipw=" crossorigin="anonymous"></script>

<!-- pkgdown -->
<link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script>



<meta property="og:title" content="Function reference" />

<meta property="og:image" content="https://scrunchy.hesselberthlab.org/logo.png" />
<meta name="twitter:card" content="summary" />



<!-- mathjax -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script>

<!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->


  </head>

  <body>
    <div class="container template-reference-index">
      <header>
      <div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">scrunchy</a>
        <span class="version label label-danger" data-toggle="tooltip" data-placement="bottom" title="Unreleased version">0.0.0.9000</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="../index.html">
    <span class="fa fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../articles/scrunchy.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul>
      
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/hesselberthlab/scrunchy">
    <span class="fa fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
      
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      
      </header>

<div class="row">
  <div class="contents col-md-9">
    <div class="page-header">
      <h1>Reference</h1>
    </div>

    <table class="ref-index">

    <colgroup>
      
      <col class="alias" />
      <col class="title" />
    </colgroup>

    <tbody>
      <tr>
        <th colspan="2">
          <h2 id="section-create-objects" class="hasAnchor"><a href="#section-create-objects" class="anchor"></a>Create Objects</h2>
          <p class="section-desc"></p>
        </th>
      </tr>
      <tr>
        
        <td>
          <p><code><a href="create_fsce.html">create_fsce()</a></code> </p>
        </td>
        <td><p>Create a functional single-cell experiment</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="create_sce_haircut.html">create_sce_haircut()</a></code> </p>
        </td>
        <td><p>Create a single-cell Haircut experiment</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="create_sce_rnaseq.html">create_sce_rnaseq()</a></code> </p>
        </td>
        <td><p>Create a single-cell mRNA-seq experiment</p></td>
      </tr>
    </tbody><tbody>
      <tr>
        <th colspan="2">
          <h2 id="section-manipulate-and-analyze-data" class="hasAnchor"><a href="#section-manipulate-and-analyze-data" class="anchor"></a>Manipulate and Analyze Data</h2>
          <p class="section-desc"></p>
        </th>
      </tr>
      <tr>
        
        <td>
          <p><code><a href="calc_hairpin_coverage.html">calc_hairpin_coverage()</a></code> </p>
        </td>
        <td><p>Calculate coverage across hairpins</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="calc_kmeans.html">calc_kmeans()</a></code> </p>
        </td>
        <td><p>K-means clustering</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="calc_pca.html">calc_pca()</a></code> </p>
        </td>
        <td><p>Calculate principal components using irlba</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="calc_tsne.html">calc_tsne()</a></code> </p>
        </td>
        <td><p>Generate 2D cell embeddings using tSNE</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="calc_umap.html">calc_umap()</a></code> </p>
        </td>
        <td><p>Generate 2D cell embeddings using UMAP</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="calc_var_features.html">calc_var_features()</a></code> </p>
        </td>
        <td><p>Get variable features</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="stat_activity_grouped.html">stat_activity_grouped()</a></code> </p>
        </td>
        <td><p>Compare activity differences between groups.</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="stat_anova_grouped.html">stat_anova_grouped()</a></code> </p>
        </td>
        <td><p>Analysis of variance of activities across groups</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="stat_anova_tukey.html">stat_anova_tukey()</a></code> </p>
        </td>
        <td><p>Post-hoc analysis of ANOVA results</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="tidy_coldata.html">tidy_coldata()</a></code> </p>
        </td>
        <td><p>Tidy colData</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="tidy_counts.html">tidy_counts()</a></code> </p>
        </td>
        <td><p>Tidy counts data</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="tidy_dims.html">tidy_dims()</a></code> </p>
        </td>
        <td><p>Tidy reducedDims data</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="tidy_logcounts.html">tidy_logcounts()</a></code> </p>
        </td>
        <td><p>Tidy logcounts data</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="clr_normalize.html">clr_normalize()</a></code> </p>
        </td>
        <td><p>Centered log-ratio normalize a matrix</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="log_normalize.html">log_normalize()</a></code> </p>
        </td>
        <td><p>Log-normalize a matrix</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="normalize.html">normalize()</a></code> </p>
        </td>
        <td><p>Normalize data in a SingleCellExperiment</p></td>
      </tr>
    </tbody><tbody>
      <tr>
        <th colspan="2">
          <h2 id="section-visualize-data" class="hasAnchor"><a href="#section-visualize-data" class="anchor"></a>Visualize Data</h2>
          <p class="section-desc"></p>
        </th>
      </tr>
      <tr>
        
        <td>
          <p><code><a href="plot_activity.html">plot_activity()</a></code> </p>
        </td>
        <td><p>Plot activities per cluster</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="plot_barcodes.html">plot_barcodes()</a></code> </p>
        </td>
        <td><p>Plot UMI count distribution for barcodes</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="plot_dims.html">plot_dims()</a></code> </p>
        </td>
        <td><p>Scatter plot of cells in a two-dimensional embedding.</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="plot_hairpin_coverage.html">plot_hairpin_coverage()</a></code> </p>
        </td>
        <td><p>Plot coverage across hairpins</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="plot_heatmap.html">plot_heatmap()</a></code> </p>
        </td>
        <td><p>Heatmap of signals</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="plot_saturation.html">plot_saturation()</a></code> </p>
        </td>
        <td><p>Plot sequencing saturation</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="palette_OkabeIto.html">palette_OkabeIto</a></code> <code><a href="palette_OkabeIto.html">palette_OkabeIto_black</a></code> </p>
        </td>
        <td><p>Color palette proposed by Okabe and Ito</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="scale_OkabeIto.html">scale_color_OkabeIto()</a></code> <code><a href="scale_OkabeIto.html">scale_fill_OkabeIto()</a></code> <code><a href="scale_OkabeIto.html">scale_OkabeIto()</a></code> </p>
        </td>
        <td><p>Okabe-Ito color scale</p></td>
      </tr>
    </tbody><tbody>
      <tr>
        <th colspan="2">
          <h2 id="section-data-sets" class="hasAnchor"><a href="#section-data-sets" class="anchor"></a>Data Sets</h2>
          <p class="section-desc"></p>
        </th>
      </tr>
      <tr>
        
        <td>
          <p><code><a href="fsce_small.html">fsce_small</a></code> </p>
        </td>
        <td><p>An example FunctionalSingleCellExperiment</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="fsce_tidy.html">fsce_tidy</a></code> </p>
        </td>
        <td><p>Tidied tibble of data from fsce_small</p></td>
      </tr>
    </tbody><tbody>
      <tr>
        <th colspan="2">
          <h2 id="section-object-definitions" class="hasAnchor"><a href="#section-object-definitions" class="anchor"></a>Object Definitions</h2>
          <p class="section-desc"></p>
        </th>
      </tr>
      <tr>
        
        <td>
          <p><code><a href="FunctionalSingleCellExperiment.html">FunctionalSingleCellExperiment()</a></code> </p>
        </td>
        <td><p>Constructor for a <code>FunctionalSingleCellExperiment</code> object</p></td>
      </tr>
    </tbody><tbody>
      <tr>
        <th colspan="2">
          <h2 id="section-utilities" class="hasAnchor"><a href="#section-utilities" class="anchor"></a>Utilities</h2>
          <p class="section-desc"></p>
        </th>
      </tr>
      <tr>
        
        <td>
          <p><code><a href="molecule_df.html">molecule_df()</a></code> </p>
        </td>
        <td><p>data.frame that contains molecule info</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="is.molecule_df.html">is.molecule_df()</a></code> </p>
        </td>
        <td><p>Test if the object is a molecule_df.</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="filter_molecules.html">filter_molecules()</a></code> </p>
        </td>
        <td><p>Filter barcodes in molecule file and write to disk</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="umitools_to_mtx.html">umitools_to_mtx()</a></code> </p>
        </td>
        <td><p>Convert umitools flat format tsv to sparseMatrix .mtx format</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="read_matrix.html">read_matrix()</a></code> </p>
        </td>
        <td><p>Load mtx formatted 10x or haircut matrices</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="read_molecules.html">read_molecules()</a></code> </p>
        </td>
        <td><p>Read molecule file into memory</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="filter_matrix.html">filter_matrix()</a></code> </p>
        </td>
        <td><p>Filter and write a sparseMatrix keeping only specified barcodes</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="write_matrix.html">write_matrix()</a></code> </p>
        </td>
        <td><p>Write a sparseMatrix to disk</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="scrunchy.html">scrunchy</a></code> </p>
        </td>
        <td><p>scrunchy: single-cell reconstruction of functional heterogeneity</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="scrunchy_data.html">scrunchy_data()</a></code> </p>
        </td>
        <td><p>Provide path to scrunchy internal data</p></td>
      </tr>
    </tbody>
    </table>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="sidebar">
    <h2>Contents</h2>
    <ul class="nav nav-pills nav-stacked">
      <li><a href="#section-create-objects">Create Objects</a></li>
      <li><a href="#section-manipulate-and-analyze-data">Manipulate and Analyze Data</a></li>
      <li><a href="#section-visualize-data">Visualize Data</a></li>
      <li><a href="#section-data-sets">Data Sets</a></li>
      <li><a href="#section-object-definitions">Object Definitions</a></li>
      <li><a href="#section-utilities">Utilities</a></li>
    </ul>
  </div>
</div>

      <footer>
      <div class="copyright">
  <p>Developed by <a href='https://hesselberthlab.org'>Jay Hesselberth</a>, Amanda Richer, Kent Riemondy.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.3.0.</p>
</div>
      </footer>
   </div>

  

  </body>
</html>

