<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Introduction to scrunchy â€¢ scrunchy</title>
<!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png">
<link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png">
<link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png">
<link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png">
<link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png">
<link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png">
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.3.1/jquery.min.js" integrity="sha256-FgpCb/KJQlLNfOu91ta32o/NMZxltwRo8QtmkMRdAu8=" crossorigin="anonymous"></script><!-- Bootstrap --><link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.3.7/yeti/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.3.7/js/bootstrap.min.js" integrity="sha256-U5ZEeKfGNOja007MMD3YBI0A3OSZOQbeG6z2f2Y0hu8=" crossorigin="anonymous"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/4.7.0/css/font-awesome.min.css" integrity="sha256-eZrrJcwDc/3uDhsdt61sL2oOBY362qM3lon1gyExkL0=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.4/clipboard.min.js" integrity="sha256-FiZwavyI2V6+EXO1U+xzLG3IKldpiTFf3153ea9zikQ=" crossorigin="anonymous"></script><!-- sticky kit --><script src="https://cdnjs.cloudflare.com/ajax/libs/sticky-kit/1.1.3/sticky-kit.min.js" integrity="sha256-c4Rlo1ZozqTPE2RLuvbusY3+SU1pQaJC0TjuhygMipw=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="Introduction to scrunchy">
<meta property="og:description" content="">
<meta property="og:image" content="https://scrunchy.hesselberthlab.org/logo.png">
<meta name="twitter:card" content="summary">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">scrunchy</a>
        <span class="version label label-danger" data-toggle="tooltip" data-placement="bottom" title="Unreleased version">0.0.0.9000</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fa fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../articles/scrunchy.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/hesselberthlab/scrunchy">
    <span class="fa fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      
      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1>Introduction to scrunchy</h1>
            
      
      <small class="dont-index">Source: <a href="https://github.com/hesselberthlab/scrunchy/blob/master/vignettes/scrunchy.Rmd"><code>vignettes/scrunchy.Rmd</code></a></small>
      <div class="hidden name"><code>scrunchy.Rmd</code></div>

    </div>

    
    
<div id="create-a-functional-single-cell-experiment-fsce" class="section level1">
<h1 class="hasAnchor">
<a href="#create-a-functional-single-cell-experiment-fsce" class="anchor"></a>Create a Functional Single Cell Experiment (fsce)</h1>
<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb1-1" data-line-number="1"><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/library">library</a></span>(scrunchy)</a>
<a class="sourceLine" id="cb1-2" data-line-number="2"></a>
<a class="sourceLine" id="cb1-3" data-line-number="3"><span class="co"># this same object is available as `fsce_small` in the package.</span></a>
<a class="sourceLine" id="cb1-4" data-line-number="4">fsce &lt;-<span class="st"> </span><span class="kw"><a href="../reference/create_fsce.html">create_fsce</a></span>(</a>
<a class="sourceLine" id="cb1-5" data-line-number="5">  <span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/list">list</a></span>(</a>
<a class="sourceLine" id="cb1-6" data-line-number="6">    <span class="dt">rnaseq =</span> <span class="kw"><a href="../reference/create_sce_rnaseq.html">create_sce_rnaseq</a></span>(<span class="kw"><a href="../reference/scrunchy_data.html">scrunchy_data</a></span>(<span class="st">"mrna"</span>)),</a>
<a class="sourceLine" id="cb1-7" data-line-number="7">    <span class="dt">haircut =</span> <span class="kw"><a href="../reference/create_sce_haircut.html">create_sce_haircut</a></span>(<span class="kw"><a href="../reference/scrunchy_data.html">scrunchy_data</a></span>(<span class="st">"haircut"</span>))</a>
<a class="sourceLine" id="cb1-8" data-line-number="8">  )</a>
<a class="sourceLine" id="cb1-9" data-line-number="9">)</a>
<a class="sourceLine" id="cb1-10" data-line-number="10"></a>
<a class="sourceLine" id="cb1-11" data-line-number="11">fsce</a></code></pre></div>
<pre><code>## A FunctionalSingleCellExperiment object of 2 listed
##  experiments with user-defined names and respective classes. 
##  Containing an ExperimentList class object of length 2: 
##  [1] rnaseq: SingleCellExperiment with 9479 rows and 250 columns 
##  [2] haircut: SingleCellExperiment with 426 rows and 250 columns 
## Features: 
##  experiments() - obtain the ExperimentList instance 
##  colData() - the primary/phenotype DataFrame 
##  sampleMap() - the sample availability DataFrame 
##  `$`, `[`, `[[` - extract colData columns, subset, or experiment 
##  *Format() - convert into a long or wide DataFrame 
##  assays() - convert ExperimentList to a SimpleList of matrices</code></pre>
<p><code>fsce</code> is an object of class <code>FunctionalSingleCellExperiment</code>, which can contain data from multiple modalities and does not require that each modality has the same number of cells or features.</p>
<div class="sourceCode" id="cb3"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb3-1" data-line-number="1"><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/library">library</a></span>(SingleCellExperiment)</a></code></pre></div>
<pre><code>## Loading required package: SummarizedExperiment</code></pre>
<pre><code>## Loading required package: GenomicRanges</code></pre>
<pre><code>## Loading required package: stats4</code></pre>
<pre><code>## Loading required package: BiocGenerics</code></pre>
<pre><code>## Loading required package: parallel</code></pre>
<pre><code>## 
## Attaching package: 'BiocGenerics'</code></pre>
<pre><code>## The following objects are masked from 'package:parallel':
## 
##     clusterApply, clusterApplyLB, clusterCall, clusterEvalQ,
##     clusterExport, clusterMap, parApply, parCapply, parLapply,
##     parLapplyLB, parRapply, parSapply, parSapplyLB</code></pre>
<pre><code>## The following object is masked from 'package:scrunchy':
## 
##     normalize</code></pre>
<pre><code>## The following objects are masked from 'package:stats':
## 
##     IQR, mad, sd, var, xtabs</code></pre>
<pre><code>## The following objects are masked from 'package:base':
## 
##     anyDuplicated, append, as.data.frame, basename, cbind,
##     colMeans, colnames, colSums, dirname, do.call, duplicated,
##     eval, evalq, Filter, Find, get, grep, grepl, intersect,
##     is.unsorted, lapply, lengths, Map, mapply, match, mget, order,
##     paste, pmax, pmax.int, pmin, pmin.int, Position, rank, rbind,
##     Reduce, rowMeans, rownames, rowSums, sapply, setdiff, sort,
##     table, tapply, union, unique, unsplit, which, which.max,
##     which.min</code></pre>
<pre><code>## Loading required package: S4Vectors</code></pre>
<pre><code>## 
## Attaching package: 'S4Vectors'</code></pre>
<pre><code>## The following object is masked from 'package:base':
## 
##     expand.grid</code></pre>
<pre><code>## Loading required package: IRanges</code></pre>
<pre><code>## Loading required package: GenomeInfoDb</code></pre>
<pre><code>## Loading required package: Biobase</code></pre>
<pre><code>## Welcome to Bioconductor
## 
##     Vignettes contain introductory material; view with
##     'browseVignettes()'. To cite Bioconductor, see
##     'citation("Biobase")', and for packages 'citation("pkgname")'.</code></pre>
<pre><code>## Loading required package: DelayedArray</code></pre>
<pre><code>## Loading required package: matrixStats</code></pre>
<pre><code>## 
## Attaching package: 'matrixStats'</code></pre>
<pre><code>## The following objects are masked from 'package:Biobase':
## 
##     anyMissing, rowMedians</code></pre>
<pre><code>## Loading required package: BiocParallel</code></pre>
<pre><code>## 
## Attaching package: 'DelayedArray'</code></pre>
<pre><code>## The following objects are masked from 'package:matrixStats':
## 
##     colMaxs, colMins, colRanges, rowMaxs, rowMins, rowRanges</code></pre>
<pre><code>## The following objects are masked from 'package:base':
## 
##     aperm, apply</code></pre>
<div class="sourceCode" id="cb29"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb29-1" data-line-number="1"><span class="co"># `rnaseq` stores RNA info</span></a>
<a class="sourceLine" id="cb29-2" data-line-number="2">fsce[[<span class="st">"rnaseq"</span>]]</a></code></pre></div>
<pre><code>## class: SingleCellExperiment 
## dim: 9479 250 
## metadata(0):
## assays(2): counts logcounts
## rownames(9479): FO538757.2 AP006222.2 ... AC004556.1 AC240274.1
## rowData names(0):
## colnames(250): TGCGGGTTCAACTCTT TGGTTCCTCCCAAGAT ...
##   CTCGAAACACACTGCG TGACGGCAGAGTCGGT
## colData names(1): cell_id
## reducedDimNames(0):
## spikeNames(0):</code></pre>
<div class="sourceCode" id="cb31"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb31-1" data-line-number="1"><span class="kw"><a href="https://www.rdocumentation.org/packages/SingleCellExperiment/topics/SCE-assays">counts</a></span>(fsce[[<span class="st">"rnaseq"</span>]])[<span class="dv">1</span><span class="op">:</span><span class="dv">5</span>, <span class="dv">1</span><span class="op">:</span><span class="dv">5</span>]</a></code></pre></div>
<pre><code>## 5 x 5 sparse Matrix of class "dgTMatrix"
##            TGCGGGTTCAACTCTT TGGTTCCTCCCAAGAT CATATTCCATGCCCGA
## FO538757.2                .                .                .
## AP006222.2                .                .                .
## NOC2L                     .                .                .
## ISG15                     .                .                .
## TNFRSF18                  .                .                .
##            TCACAAGTCTCCAACC GGACAAGAGCCCTAAT
## FO538757.2                1                .
## AP006222.2                .                .
## NOC2L                     .                .
## ISG15                     .                .
## TNFRSF18                  .                .</code></pre>
<div class="sourceCode" id="cb33"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb33-1" data-line-number="1"><span class="co"># `haircut` stores functional data</span></a>
<a class="sourceLine" id="cb33-2" data-line-number="2">fsce[[<span class="st">"haircut"</span>]]</a></code></pre></div>
<pre><code>## class: SingleCellExperiment 
## dim: 426 250 
## metadata(0):
## assays(2): counts logcounts
## rownames(426): Abasic_1 Abasic_10 ... riboG_8 riboG_9
## rowData names(2): hairpin position
## colnames(250): TGCGGGTTCAACTCTT TGGTTCCTCCCAAGAT ...
##   CTCGAAACACACTGCG TGACGGCAGAGTCGGT
## colData names(1): cell_id
## reducedDimNames(0):
## spikeNames(0):</code></pre>
<div class="sourceCode" id="cb35"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb35-1" data-line-number="1"><span class="kw"><a href="https://www.rdocumentation.org/packages/SingleCellExperiment/topics/SCE-assays">counts</a></span>(fsce[[<span class="st">"haircut"</span>]])[<span class="dv">1</span><span class="op">:</span><span class="dv">5</span>, <span class="dv">1</span><span class="op">:</span><span class="dv">5</span>]</a></code></pre></div>
<pre><code>## 5 x 5 sparse Matrix of class "dgTMatrix"
##           TGCGGGTTCAACTCTT TGGTTCCTCCCAAGAT CATATTCCATGCCCGA
## Abasic_1                 8               17               11
## Abasic_10                .                .                .
## Abasic_11                .                .                .
## Abasic_12                .                .                .
## Abasic_13                .                1                .
##           TCACAAGTCTCCAACC GGACAAGAGCCCTAAT
## Abasic_1                 7               18
## Abasic_10                .                .
## Abasic_11                .                .
## Abasic_12                .                .
## Abasic_13                .                .</code></pre>
<div class="sourceCode" id="cb37"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb37-1" data-line-number="1"><span class="co"># row annotations contain info about each row in the matrix</span></a>
<a class="sourceLine" id="cb37-2" data-line-number="2"><span class="kw">rowData</span>(fsce[[<span class="st">"haircut"</span>]])</a></code></pre></div>
<pre><code>## DataFrame with 426 rows and 2 columns
##               hairpin    position
##           &lt;character&gt; &lt;character&gt;
## Abasic_1       Abasic           1
## Abasic_10      Abasic          10
## Abasic_11      Abasic          11
## Abasic_12      Abasic          12
## Abasic_13      Abasic          13
## ...               ...         ...
## riboG_60        riboG          60
## riboG_61        riboG          61
## riboG_7         riboG           7
## riboG_8         riboG           8
## riboG_9         riboG           9</code></pre>
<div class="sourceCode" id="cb39"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb39-1" data-line-number="1"><span class="co"># column annotations contain info about each row in the matrix, </span></a>
<a class="sourceLine" id="cb39-2" data-line-number="2"><span class="co"># or about each unique cell present in any of the experiments</span></a>
<a class="sourceLine" id="cb39-3" data-line-number="3"><span class="kw">colData</span>(fsce)</a></code></pre></div>
<pre><code>## DataFrame with 250 rows and 1 column
##                           cell_id
##                       &lt;character&gt;
## TGCGGGTTCAACTCTT TGCGGGTTCAACTCTT
## TGGTTCCTCCCAAGAT TGGTTCCTCCCAAGAT
## CATATTCCATGCCCGA CATATTCCATGCCCGA
## TCACAAGTCTCCAACC TCACAAGTCTCCAACC
## GGACAAGAGCCCTAAT GGACAAGAGCCCTAAT
## ...                           ...
## CTACGTCCAGTGGAGT CTACGTCCAGTGGAGT
## TTAGGCACAGGTGGAT TTAGGCACAGGTGGAT
## AGCGTCGTCGTCGTTC AGCGTCGTCGTCGTTC
## CTCGAAACACACTGCG CTCGAAACACACTGCG
## TGACGGCAGAGTCGGT TGACGGCAGAGTCGGT</code></pre>
</div>
<div id="plots" class="section level1">
<h1 class="hasAnchor">
<a href="#plots" class="anchor"></a>Plots</h1>
<p><code>plot_hairpin</code> produces a coverage plot across each hairpin in the data.</p>
<div class="sourceCode" id="cb41"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb41-1" data-line-number="1"><span class="kw"><a href="../reference/plot_hairpin_coverage.html">plot_hairpin_coverage</a></span>(fsce)</a></code></pre></div>
<p><img src="scrunchy_files/figure-html/hairpin_plot-1.png" width="100%"></p>
</div>
<div id="normalization" class="section level1">
<h1 class="hasAnchor">
<a href="#normalization" class="anchor"></a>Normalization</h1>
<p>Normalized data is calculated by the <code>create_</code> functions.</p>
<p>RNA data is log-normalized and scaled to total UMI count for each cell. Functional data is normalized using the center log ratio method used for CITE-seq analysis.</p>
<p>Normalized data is stored in the <code>logcounts</code> slot of each <code>SingleCellExperiment</code> object.</p>
<div class="sourceCode" id="cb42"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb42-1" data-line-number="1"><span class="kw"><a href="https://www.rdocumentation.org/packages/SingleCellExperiment/topics/SCE-assays">logcounts</a></span>(fsce[[<span class="st">"haircut"</span>]])[<span class="dv">1</span><span class="op">:</span><span class="dv">5</span>, <span class="dv">1</span><span class="op">:</span><span class="dv">3</span>]</a></code></pre></div>
<pre><code>##           TGCGGGTTCAACTCTT TGGTTCCTCCCAAGAT CATATTCCATGCCCGA
## Abasic_1          1.614359        2.1843851         1.931199
## Abasic_10         0.000000        0.0000000         0.000000
## Abasic_11         0.000000        0.0000000         0.000000
## Abasic_12         0.000000        0.0000000         0.000000
## Abasic_13         0.000000        0.3810593         0.000000</code></pre>
</div>
<div id="feature-selection-and-pca" class="section level1">
<h1 class="hasAnchor">
<a href="#feature-selection-and-pca" class="anchor"></a>Feature selection and PCA</h1>
<div class="sourceCode" id="cb44"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb44-1" data-line-number="1"><span class="co"># get variable genes</span></a>
<a class="sourceLine" id="cb44-2" data-line-number="2">var_genes &lt;-<span class="st"> </span><span class="kw"><a href="../reference/calc_var_features.html">calc_var_features</a></span>(fsce, <span class="dt">n =</span> <span class="dv">1000</span>) </a>
<a class="sourceLine" id="cb44-3" data-line-number="3"></a>
<a class="sourceLine" id="cb44-4" data-line-number="4"><span class="co"># calc PCA and embed in 2D</span></a>
<a class="sourceLine" id="cb44-5" data-line-number="5">fsce &lt;-<span class="st"> </span><span class="kw"><a href="../reference/calc_pca.html">calc_pca</a></span>(fsce, <span class="dt">n_pcs =</span> <span class="dv">20</span>, <span class="dt">genes =</span> var_genes)</a></code></pre></div>
<pre><code>## scaling data</code></pre>
<pre><code>## calculating pcs</code></pre>
</div>
<div id="embed-cells-in-2d-using-umap-or-tsne" class="section level1">
<h1 class="hasAnchor">
<a href="#embed-cells-in-2d-using-umap-or-tsne" class="anchor"></a>Embed cells in 2D using UMAP or tSNE</h1>
<div class="sourceCode" id="cb47"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb47-1" data-line-number="1">fsce &lt;-<span class="st"> </span><span class="kw"><a href="../reference/calc_umap.html">calc_umap</a></span>(fsce, <span class="dt">n_dims =</span> <span class="dv">6</span>)</a>
<a class="sourceLine" id="cb47-2" data-line-number="2"></a>
<a class="sourceLine" id="cb47-3" data-line-number="3">fsce &lt;-<span class="st"> </span><span class="kw"><a href="../reference/calc_tsne.html">calc_tsne</a></span>(fsce, <span class="dt">n_dims =</span> <span class="dv">6</span>)</a></code></pre></div>
<p>Cell embeddings are stored in the <code>reducedDims</code> slot of the <code>SingleCellExperiment</code> and can be accessed as a named list.</p>
<div class="sourceCode" id="cb48"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb48-1" data-line-number="1"><span class="kw"><a href="https://www.rdocumentation.org/packages/SingleCellExperiment/topics/SCE-reducedDims">reducedDims</a></span>(fsce[[<span class="st">"rnaseq"</span>]])</a></code></pre></div>
<pre><code>## List of length 3
## names(3): PCA UMAP TSNE</code></pre>
<div class="sourceCode" id="cb50"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb50-1" data-line-number="1"><span class="kw"><a href="https://www.rdocumentation.org/packages/SingleCellExperiment/topics/SCE-reducedDims">reducedDims</a></span>(fsce[[<span class="st">"rnaseq"</span>]])<span class="op">$</span>UMAP[<span class="dv">1</span><span class="op">:</span><span class="dv">4</span>, ]</a></code></pre></div>
<pre><code>##                        [,1]       [,2]
## TGCGGGTTCAACTCTT -0.7056742 -3.1788763
## TGGTTCCTCCCAAGAT  0.0289938 -3.8062140
## CATATTCCATGCCCGA  0.1522357 -0.9630351
## TCACAAGTCTCCAACC  4.5660902 -0.5486145</code></pre>
</div>
<div id="tidying-data-for-analysis-and-plotting" class="section level1">
<h1 class="hasAnchor">
<a href="#tidying-data-for-analysis-and-plotting" class="anchor"></a>Tidying data for analysis and plotting</h1>
<p>Several <code>tidy_</code> functions are provided to convert <code>fsce</code> into a tidy formate.</p>
<div class="sourceCode" id="cb52"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb52-1" data-line-number="1"><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/library">library</a></span>(dplyr)</a></code></pre></div>
<pre><code>## 
## Attaching package: 'dplyr'</code></pre>
<pre><code>## The following object is masked from 'package:matrixStats':
## 
##     count</code></pre>
<pre><code>## The following object is masked from 'package:Biobase':
## 
##     combine</code></pre>
<pre><code>## The following objects are masked from 'package:GenomicRanges':
## 
##     intersect, setdiff, union</code></pre>
<pre><code>## The following object is masked from 'package:GenomeInfoDb':
## 
##     intersect</code></pre>
<pre><code>## The following objects are masked from 'package:IRanges':
## 
##     collapse, desc, intersect, setdiff, slice, union</code></pre>
<pre><code>## The following objects are masked from 'package:S4Vectors':
## 
##     first, intersect, rename, setdiff, setequal, union</code></pre>
<pre><code>## The following objects are masked from 'package:BiocGenerics':
## 
##     combine, intersect, setdiff, union</code></pre>
<pre><code>## The following objects are masked from 'package:stats':
## 
##     filter, lag</code></pre>
<pre><code>## The following objects are masked from 'package:base':
## 
##     intersect, setdiff, setequal, union</code></pre>
<div class="sourceCode" id="cb63"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb63-1" data-line-number="1"><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/library">library</a></span>(purrr)</a></code></pre></div>
<pre><code>## 
## Attaching package: 'purrr'</code></pre>
<pre><code>## The following object is masked from 'package:DelayedArray':
## 
##     simplify</code></pre>
<pre><code>## The following object is masked from 'package:GenomicRanges':
## 
##     reduce</code></pre>
<pre><code>## The following object is masked from 'package:IRanges':
## 
##     reduce</code></pre>
<div class="sourceCode" id="cb68"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb68-1" data-line-number="1">fsce_tidy &lt;-<span class="st"> </span>purrr<span class="op">::</span><span class="kw"><a href="https://purrr.tidyverse.org/reference/reduce.html">reduce</a></span>(</a>
<a class="sourceLine" id="cb68-2" data-line-number="2">  <span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/list">list</a></span>(</a>
<a class="sourceLine" id="cb68-3" data-line-number="3">    <span class="kw"><a href="../reference/tidy_dims.html">tidy_dims</a></span>(fsce_small) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb68-4" data-line-number="4"><span class="st">      </span><span class="kw"><a href="https://dplyr.tidyverse.org/reference/select.html">select</a></span>(cell_id, <span class="kw"><a href="https://dplyr.tidyverse.org/reference/reexports.html">starts_with</a></span>(<span class="st">"UMAP"</span>), <span class="op">-</span>experiment),</a>
<a class="sourceLine" id="cb68-5" data-line-number="5">    <span class="kw"><a href="../reference/tidy_coldata.html">tidy_coldata</a></span>(fsce_small),</a>
<a class="sourceLine" id="cb68-6" data-line-number="6">    <span class="kw"><a href="../reference/tidy_logcounts.html">tidy_logcounts</a></span>(fsce_small[<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/c">c</a></span>(<span class="st">"Uracil_45"</span>, <span class="st">"riboG_44"</span>), , ]) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb68-7" data-line-number="7"><span class="st">      </span><span class="kw"><a href="https://dplyr.tidyverse.org/reference/select.html">select</a></span>(<span class="op">-</span>experiment)</a>
<a class="sourceLine" id="cb68-8" data-line-number="8">  ),</a>
<a class="sourceLine" id="cb68-9" data-line-number="9">  left_join,</a>
<a class="sourceLine" id="cb68-10" data-line-number="10">  <span class="dt">by =</span> <span class="st">"cell_id"</span></a>
<a class="sourceLine" id="cb68-11" data-line-number="11">)</a>
<a class="sourceLine" id="cb68-12" data-line-number="12"></a>
<a class="sourceLine" id="cb68-13" data-line-number="13">fsce_tidy</a></code></pre></div>
<pre><code>## # A tibble: 250 x 6
##    cell_id           UMAP1  UMAP2 k_cluster Uracil_45 riboG_44
##    &lt;chr&gt;             &lt;dbl&gt;  &lt;dbl&gt; &lt;chr&gt;         &lt;dbl&gt;    &lt;dbl&gt;
##  1 TGCGGGTTCAACTCTT  1.53   2.75  6             1.39     1.61 
##  2 TGGTTCCTCCCAAGAT  1.04   1.82  6             1.55     1.33 
##  3 CATATTCCATGCCCGA  1.68   4.18  6             1.76     0.429
##  4 TCACAAGTCTCCAACC -2.72   2.56  6             0.418    0.418
##  5 GGACAAGAGCCCTAAT  0.945  3.07  6             2.09     1.46 
##  6 CTTTGCGAGCAGGTCA  1.07   3.94  6             1.08     1.08 
##  7 GTGCAGCAGTAGCCGA  2.25   1.55  6             2.32     1.59 
##  8 AGCGGTCCAATAGAGT -2.55  -0.766 4             1.21     1.46 
##  9 GGATGTTTCTCAAACG  1.30   2.09  6             1.49     0.400
## 10 GTACTCCTCGTTTAGG -4.90   2.47  5             1.50     0.405
## # â€¦ with 240 more rows</code></pre>
</div>
<div id="visualization" class="section level1">
<h1 class="hasAnchor">
<a href="#visualization" class="anchor"></a>Visualization</h1>
<div id="two-dimensional-embeddings" class="section level2">
<h2 class="hasAnchor">
<a href="#two-dimensional-embeddings" class="anchor"></a>Two-dimensional embeddings</h2>
<p>Features can be visulized on a two-dimensional embedding with the <code><a href="../reference/plot_dims.html">plot_dims()</a></code> function which will plot gene/function assay values.</p>
<div class="sourceCode" id="cb70"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb70-1" data-line-number="1">cowplot<span class="op">::</span><span class="kw"><a href="https://www.rdocumentation.org/packages/cowplot/topics/plot_grid">plot_grid</a></span>(</a>
<a class="sourceLine" id="cb70-2" data-line-number="2">  <span class="kw"><a href="../reference/plot_dims.html">plot_dims</a></span>(fsce_tidy, UMAP1, UMAP2, Uracil_<span class="dv">45</span>, <span class="dt">size =</span> <span class="dv">1</span>),</a>
<a class="sourceLine" id="cb70-3" data-line-number="3">  <span class="kw"><a href="../reference/plot_dims.html">plot_dims</a></span>(fsce_tidy, UMAP1, UMAP2, riboG_<span class="dv">44</span>, <span class="dt">size =</span> <span class="dv">1</span>)</a>
<a class="sourceLine" id="cb70-4" data-line-number="4">)</a></code></pre></div>
<p><img src="scrunchy_files/figure-html/plot_dims-1.png" width="100%"></p>
</div>
<div id="functional-activities" class="section level2">
<h2 class="hasAnchor">
<a href="#functional-activities" class="anchor"></a>Functional activities</h2>
<p>Activities can be visualized per-group with <code>plot_actvitiy()</code>. These two plots have uracil and ribonucleotide excision activities plotted per cluster (k-means, n = 6).</p>
<div class="sourceCode" id="cb71"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb71-1" data-line-number="1">cowplot<span class="op">::</span><span class="kw"><a href="https://www.rdocumentation.org/packages/cowplot/topics/plot_grid">plot_grid</a></span>(</a>
<a class="sourceLine" id="cb71-2" data-line-number="2">  <span class="kw"><a href="../reference/plot_activity.html">plot_activity</a></span>(fsce_tidy, Uracil_<span class="dv">45</span>, k_cluster),</a>
<a class="sourceLine" id="cb71-3" data-line-number="3">  <span class="kw"><a href="../reference/plot_activity.html">plot_activity</a></span>(fsce_tidy, riboG_<span class="dv">44</span>, k_cluster)</a>
<a class="sourceLine" id="cb71-4" data-line-number="4">)</a></code></pre></div>
<p><img src="scrunchy_files/figure-html/plot_activity-1.png" width="100%"></p>
</div>
<div id="heatmap" class="section level2">
<h2 class="hasAnchor">
<a href="#heatmap" class="anchor"></a>Heatmap</h2>
<p>These heatmaps show uracil and ribonucleotide excision activities for the U:A and riboG hairpins</p>
<div class="sourceCode" id="cb72"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb72-1" data-line-number="1">mtx &lt;-<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/SingleCellExperiment/topics/SCE-assays">logcounts</a></span>(fsce_small[[<span class="st">"haircut"</span>]])</a>
<a class="sourceLine" id="cb72-2" data-line-number="2"></a>
<a class="sourceLine" id="cb72-3" data-line-number="3">rows &lt;-<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/paste">paste</a></span>(<span class="st">"Uracil"</span>, <span class="dv">1</span><span class="op">:</span><span class="dv">61</span>, <span class="dt">sep =</span> <span class="st">"_"</span>)</a>
<a class="sourceLine" id="cb72-4" data-line-number="4"><span class="kw"><a href="../reference/plot_heatmap.html">plot_heatmap</a></span>(mtx, rows, <span class="dt">name =</span> <span class="st">"Uracil"</span>)</a></code></pre></div>
<p><img src="scrunchy_files/figure-html/heatmap_uracil-1.png" width="100%"></p>
<div class="sourceCode" id="cb73"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb73-1" data-line-number="1">rows &lt;-<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/paste">paste</a></span>(<span class="st">"riboG"</span>, <span class="dv">1</span><span class="op">:</span><span class="dv">61</span>, <span class="dt">sep =</span> <span class="st">"_"</span>)</a>
<a class="sourceLine" id="cb73-2" data-line-number="2"><span class="kw"><a href="../reference/plot_heatmap.html">plot_heatmap</a></span>(mtx, rows, <span class="dt">name =</span> <span class="st">"riboG"</span>)</a></code></pre></div>
<p><img src="scrunchy_files/figure-html/heatpmap_riboG-1.png" width="100%"></p>
</div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="sidebar">
        <div id="tocnav">
      <h2 class="hasAnchor">
<a href="#tocnav" class="anchor"></a>Contents</h2>
      <ul class="nav nav-pills nav-stacked">
<li><a href="#create-a-functional-single-cell-experiment-fsce">Create a Functional Single Cell Experiment (fsce)</a></li>
      <li><a href="#plots">Plots</a></li>
      <li><a href="#normalization">Normalization</a></li>
      <li><a href="#feature-selection-and-pca">Feature selection and PCA</a></li>
      <li><a href="#embed-cells-in-2d-using-umap-or-tsne">Embed cells in 2D using UMAP or tSNE</a></li>
      <li><a href="#tidying-data-for-analysis-and-plotting">Tidying data for analysis and plotting</a></li>
      <li>
<a href="#visualization">Visualization</a><ul class="nav nav-pills nav-stacked">
<li><a href="#two-dimensional-embeddings">Two-dimensional embeddings</a></li>
      <li><a href="#functional-activities">Functional activities</a></li>
      <li><a href="#heatmap">Heatmap</a></li>
      </ul>
</li>
      </ul>
</div>
      </div>

</div>


      <footer><div class="copyright">
  <p>Developed by <a href="https://hesselberthlab.org">Jay Hesselberth</a>, Amanda Richer, Kent Riemondy.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.3.0.</p>
</div>
      </footer>
</div>

  

  </body>
</html>
