<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Introduction to scrunchy • scrunchy</title>
<!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png">
<link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png">
<link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png">
<link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png">
<link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png">
<link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png">
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.3.1/jquery.min.js" integrity="sha256-FgpCb/KJQlLNfOu91ta32o/NMZxltwRo8QtmkMRdAu8=" crossorigin="anonymous"></script><!-- Bootstrap --><link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.3.7/sandstone/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.3.7/js/bootstrap.min.js" integrity="sha256-U5ZEeKfGNOja007MMD3YBI0A3OSZOQbeG6z2f2Y0hu8=" crossorigin="anonymous"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.7.1/css/all.min.css" integrity="sha256-nAmazAk6vS34Xqo0BSrTb+abbtFlgsFK7NKSi6o7Y78=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.7.1/css/v4-shims.min.css" integrity="sha256-6qHlizsOWFskGlwVOKuns+D1nB6ssZrHQrNj1wGplHc=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.4/clipboard.min.js" integrity="sha256-FiZwavyI2V6+EXO1U+xzLG3IKldpiTFf3153ea9zikQ=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.9.4/headroom.min.js" integrity="sha256-DJFC1kqIhelURkuza0AvYal5RxMtpzLjFhsnVIeuk+U=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.9.4/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><link href="../extra.css" rel="stylesheet">
<meta property="og:title" content="Introduction to scrunchy">
<meta property="og:description" content="">
<meta property="og:image" content="https://scrunchy.hesselberthlab.org/logo.png">
<meta name="twitter:card" content="summary">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--><!-- Global site tag (gtag.js) - Google Analytics --><script async src="https://www.googletagmanager.com/gtag/js?id=UA-54095998-8"></script><script>
  window.dataLayer = window.dataLayer || [];
  function gtag(){dataLayer.push(arguments);}
  gtag('js', new Date());

  gtag('config', 'UA-54095998-8');
</script>
</head>
<body>
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">scrunchy</a>
        <span class="version label label-danger" data-toggle="tooltip" data-placement="bottom" title="Unreleased version">0.0.0.9000</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fas fa fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../articles/scrunchy.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/hesselberthlab/scrunchy">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1>Introduction to scrunchy</h1>
            
      
      <small class="dont-index">Source: <a href="https://github.com/hesselberthlab/scrunchy/blob/master/vignettes/scrunchy.Rmd"><code>vignettes/scrunchy.Rmd</code></a></small>
      <div class="hidden name"><code>scrunchy.Rmd</code></div>

    </div>

    
    
<p>The goal of scrunchy is to facilitate the analysis of experiments that measure the heterogeneity of enzymatic activities among single cells.</p>
<p>We have developed functional assays as a new modality for single-cell experiments. These assays measure the conversion of substrates to products by single-cell extracts in a high-throghput DNA sequencing experiment. The assays are performed in the context of traditional single-cell mRNA sequencing experiment, enabling type identifcation and functional assessment of single cells. Our initial efforts have focused on measuring DNA repair activities in single cells and we are developing methods for other enzymes including proteases and kinases. Together these assays provide a new type of information from single-cell experiments.</p>
<p>The method to measure DNA repair enzymes takes advantage of the fact that nearly all DNA repair initiates with strand incision. By including synthetic polyadenylated DNA hairpins containing defined lesions, we can measure the number of incisions by cellular DNA repair factors in a high-throughput DNA sequencing experiment. Because we are measuring strand incision on DNA hairpins, we call the assay Haircut—and scrunchy keeps track of your Haircut experiments. The following figure provides an overview of the method.</p>
<p><img src="figures/method-overview.svg" width="100%"></p>
<div id="overview" class="section level2">
<h2 class="hasAnchor">
<a href="#overview" class="anchor"></a>Overview</h2>
<p>scrunchy facilitates the analysis and visualization of combined functional and transcriptome data from single cells.</p>
<ol style="list-style-type: decimal">
<li>Processing</li>
<li>Statistical analysis</li>
<li>Visualization</li>
</ol>
</div>
<div id="the-functional-single-cell-experiment-fsce" class="section level2">
<h2 class="hasAnchor">
<a href="#the-functional-single-cell-experiment-fsce" class="anchor"></a>The Functional Single Cell Experiment (fsce)</h2>
<p><code>fsce</code> is an object of class <code>FunctionalSingleCellExperiment</code>, which can contain data from multiple modalities and does not require that each modality has the same number of cells or features. It is a thin wrapper around [<code><a href="https://rdrr.io/pkg/MultiAssayExperiment/man/MultiAssayExperiment.html">MultiAssayExperiment::MultiAssayExperiment</a></code>].</p>
<p>A <code>FunctionalSingleCellExperiment</code> contains one or more objects of class [<code>SingleCellExperiment</code>], which contain observations and derived measurements that are linked by barcodes attached during sample work-up.</p>
<p>scrunchy contains a subset of cells from an experiment to measure gene expression and DNA repair activities in human peripheral blood mononuclear cells.</p>
<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb1-1" data-line-number="1"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span>(scrunchy)</a>
<a class="sourceLine" id="cb1-2" data-line-number="2">fsce_small</a></code></pre></div>
<pre><code>## A FunctionalSingleCellExperiment object of 2 listed
##  experiments with user-defined names and respective classes. 
##  Containing an ExperimentList class object of length 2: 
##  [1] rnaseq: SingleCellExperiment with 9462 rows and 250 columns 
##  [2] haircut: SingleCellExperiment with 426 rows and 250 columns 
## Features: 
##  experiments() - obtain the ExperimentList instance 
##  colData() - the primary/phenotype DataFrame 
##  sampleMap() - the sample availability DataFrame 
##  `$`, `[`, `[[` - extract colData columns, subset, or experiment 
##  *Format() - convert into a long or wide DataFrame 
##  assays() - convert ExperimentList to a SimpleList of matrices</code></pre>
<div id="individual-experiments" class="section level3">
<h3 class="hasAnchor">
<a href="#individual-experiments" class="anchor"></a>Individual experiments</h3>
<div class="sourceCode" id="cb3"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb3-1" data-line-number="1"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span>(SingleCellExperiment)</a>
<a class="sourceLine" id="cb3-2" data-line-number="2"></a>
<a class="sourceLine" id="cb3-3" data-line-number="3"><span class="co"># `rnaseq` stores RNA info</span></a>
<a class="sourceLine" id="cb3-4" data-line-number="4">fsce_small[[<span class="st">"rnaseq"</span>]]</a></code></pre></div>
<pre><code>## class: SingleCellExperiment 
## dim: 9462 250 
## metadata(1): PCA
## assays(2): counts logcounts
## rownames(9462): NOC2L HES4 ... AC004556.1 AC240274.1
## rowData names(0):
## colnames(250): TGCGGGTGTAGAGTGC TGGTTCCCATCTATGG ...
##   CTCGAAAAGTTCGATC TGACAACGTCGCATAT
## colData names(4): cell_id k_cluster leiden_cluster cell_cycle
## reducedDimNames(3): PCA UMAP TSNE
## spikeNames(0):
## altExpNames(0):</code></pre>
<div class="sourceCode" id="cb5"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb5-1" data-line-number="1"><span class="kw"><a href="https://rdrr.io/pkg/SingleCellExperiment/man/assays.html">counts</a></span>(fsce_small[[<span class="st">"rnaseq"</span>]])[<span class="dv">1</span><span class="op">:</span><span class="dv">5</span>, <span class="dv">1</span><span class="op">:</span><span class="dv">5</span>]</a></code></pre></div>
<pre><code>## 5 x 5 sparse Matrix of class "dgTMatrix"
##         TGCGGGTGTAGAGTGC TGGTTCCCATCTATGG CATATTCCACGTCAGC
## NOC2L                  .                .                .
## HES4                   .                .                .
## ISG15                  .                .                1
## TNFRSF4                .                .                .
## SDF4                   .                .                .
##         TCACAAGTCCTGCTTG GGAATAATCCAGGGCT
## NOC2L                  .                .
## HES4                   .                .
## ISG15                  .                .
## TNFRSF4                .                .
## SDF4                   .                .</code></pre>
<div class="sourceCode" id="cb7"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb7-1" data-line-number="1"><span class="co"># `haircut` stores functional data</span></a>
<a class="sourceLine" id="cb7-2" data-line-number="2">fsce_small[[<span class="st">"haircut"</span>]]</a></code></pre></div>
<pre><code>## class: SingleCellExperiment 
## dim: 426 250 
## metadata(0):
## assays(2): counts logcounts
## rownames(426): Abasic_1 Abasic_10 ... riboG_8 riboG_9
## rowData names(2): hairpin position
## colnames(250): TGCGGGTGTAGAGTGC TGGTTCCCATCTATGG ...
##   CTCGAAAAGTTCGATC TGACAACGTCGCATAT
## colData names(1): cell_id
## reducedDimNames(0):
## spikeNames(0):
## altExpNames(0):</code></pre>
<div class="sourceCode" id="cb9"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb9-1" data-line-number="1"><span class="kw"><a href="https://rdrr.io/pkg/SingleCellExperiment/man/assays.html">counts</a></span>(fsce_small[[<span class="st">"haircut"</span>]])[<span class="dv">1</span><span class="op">:</span><span class="dv">5</span>, <span class="dv">1</span><span class="op">:</span><span class="dv">5</span>]</a></code></pre></div>
<pre><code>## 5 x 5 sparse Matrix of class "dgTMatrix"
##           TGCGGGTGTAGAGTGC TGGTTCCCATCTATGG CATATTCCACGTCAGC
## Abasic_1                13               21               14
## Abasic_10                .                .                .
## Abasic_11                1                .                .
## Abasic_12                .                .                .
## Abasic_13                .                .                .
##           TCACAAGTCCTGCTTG GGAATAATCCAGGGCT
## Abasic_1                 7               14
## Abasic_10                .                .
## Abasic_11                .                .
## Abasic_12                .                .
## Abasic_13                .                .</code></pre>
<div class="sourceCode" id="cb11"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb11-1" data-line-number="1"><span class="co"># row annotations contain info about each row in the matrix</span></a>
<a class="sourceLine" id="cb11-2" data-line-number="2"><span class="kw">rowData</span>(fsce_small[[<span class="st">"haircut"</span>]])</a></code></pre></div>
<pre><code>## DataFrame with 426 rows and 2 columns
##               hairpin    position
##           &lt;character&gt; &lt;character&gt;
## Abasic_1       Abasic           1
## Abasic_10      Abasic          10
## Abasic_11      Abasic          11
## Abasic_12      Abasic          12
## Abasic_13      Abasic          13
## ...               ...         ...
## riboG_60        riboG          60
## riboG_61        riboG          61
## riboG_7         riboG           7
## riboG_8         riboG           8
## riboG_9         riboG           9</code></pre>
<div class="sourceCode" id="cb13"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb13-1" data-line-number="1"><span class="co"># column annotations contain info about each row in the matrix, </span></a>
<a class="sourceLine" id="cb13-2" data-line-number="2"><span class="co"># or about each unique cell present in any of the experiments</span></a>
<a class="sourceLine" id="cb13-3" data-line-number="3"><span class="kw">colData</span>(fsce_small)</a></code></pre></div>
<pre><code>## DataFrame with 250 rows and 1 column
##                           cell_id
##                       &lt;character&gt;
## TGCGGGTGTAGAGTGC TGCGGGTGTAGAGTGC
## TGGTTCCCATCTATGG TGGTTCCCATCTATGG
## CATATTCCACGTCAGC CATATTCCACGTCAGC
## TCACAAGTCCTGCTTG TCACAAGTCCTGCTTG
## GGAATAATCCAGGGCT GGAATAATCCAGGGCT
## ...                           ...
## CTACGTCCACCACGTG CTACGTCCACCACGTG
## TGTGGTAAGGTGCTTT TGTGGTAAGGTGCTTT
## AGCGTCGGTCGAGTTT AGCGTCGGTCGAGTTT
## CTCGAAAAGTTCGATC CTCGAAAAGTTCGATC
## TGACAACGTCGCATAT TGACAACGTCGCATAT</code></pre>
</div>
</div>
<div id="plots" class="section level1">
<h1 class="hasAnchor">
<a href="#plots" class="anchor"></a>Plots</h1>
<p><code>plot_hairpin</code> produces a coverage plot across each hairpin in the data.</p>
<div class="sourceCode" id="cb15"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb15-1" data-line-number="1"><span class="kw"><a href="../reference/plot_hairpin_coverage.html">plot_hairpin_coverage</a></span>(fsce_small)</a></code></pre></div>
<p><img src="scrunchy_files/figure-html/hairpin_plot-1.png" width="100%"></p>
</div>
<div id="normalization" class="section level1">
<h1 class="hasAnchor">
<a href="#normalization" class="anchor"></a>Normalization</h1>
<p>Normalized data is calculated by the <code>create_</code> functions.</p>
<p>RNA data is log-normalized and scaled to total UMI count for each cell. Functional data is normalized using the center log ratio method used for CITE-seq analysis.</p>
<p>Normalized data is stored in the <code>logcounts</code> slot of each <code>SingleCellExperiment</code> object.</p>
<div class="sourceCode" id="cb16"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb16-1" data-line-number="1"><span class="kw"><a href="https://rdrr.io/pkg/SingleCellExperiment/man/assays.html">logcounts</a></span>(fsce_small[[<span class="st">"haircut"</span>]])[<span class="dv">1</span><span class="op">:</span><span class="dv">5</span>, <span class="dv">1</span><span class="op">:</span><span class="dv">3</span>]</a></code></pre></div>
<pre><code>##           TGCGGGTGTAGAGTGC TGGTTCCCATCTATGG CATATTCCACGTCAGC
## Abasic_1         1.9849896         2.351059         2.053252
## Abasic_10        0.0000000         0.000000         0.000000
## Abasic_11        0.3940656         0.000000         0.000000
## Abasic_12        0.0000000         0.000000         0.000000
## Abasic_13        0.0000000         0.000000         0.000000</code></pre>
</div>
<div id="feature-selection-and-pca" class="section level1">
<h1 class="hasAnchor">
<a href="#feature-selection-and-pca" class="anchor"></a>Feature selection and PCA</h1>
<div class="sourceCode" id="cb18"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb18-1" data-line-number="1"><span class="co"># get variable genes</span></a>
<a class="sourceLine" id="cb18-2" data-line-number="2">var_genes &lt;-<span class="st"> </span><span class="kw"><a href="../reference/calc_var_features.html">calc_var_features</a></span>(fsce_small, <span class="dt">n =</span> <span class="dv">1000</span>) </a>
<a class="sourceLine" id="cb18-3" data-line-number="3"></a>
<a class="sourceLine" id="cb18-4" data-line-number="4"><span class="co"># calc PCA and embed in 2D</span></a>
<a class="sourceLine" id="cb18-5" data-line-number="5">fsce &lt;-<span class="st"> </span><span class="kw"><a href="../reference/calc_pca.html">calc_pca</a></span>(fsce_small, <span class="dt">n_pcs =</span> <span class="dv">20</span>, <span class="dt">genes =</span> var_genes)</a></code></pre></div>
<pre><code>## scaling data</code></pre>
<pre><code>## calculating pcs</code></pre>
</div>
<div id="embed-cells-in-2d-using-umap-or-tsne" class="section level1">
<h1 class="hasAnchor">
<a href="#embed-cells-in-2d-using-umap-or-tsne" class="anchor"></a>Embed cells in 2D using UMAP or tSNE</h1>
<div class="sourceCode" id="cb21"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb21-1" data-line-number="1">fsce &lt;-<span class="st"> </span><span class="kw"><a href="../reference/calc_umap.html">calc_umap</a></span>(fsce_small, <span class="dt">n_dims =</span> <span class="dv">6</span>)</a>
<a class="sourceLine" id="cb21-2" data-line-number="2"></a>
<a class="sourceLine" id="cb21-3" data-line-number="3">fsce &lt;-<span class="st"> </span><span class="kw"><a href="../reference/calc_tsne.html">calc_tsne</a></span>(fsce_small, <span class="dt">n_dims =</span> <span class="dv">6</span>)</a></code></pre></div>
<p>Cell embeddings are stored in the <code>reducedDims</code> slot of the <code>SingleCellExperiment</code> and can be accessed as a named list.</p>
<div class="sourceCode" id="cb22"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb22-1" data-line-number="1"><span class="kw"><a href="https://rdrr.io/pkg/SingleCellExperiment/man/reducedDims.html">reducedDims</a></span>(fsce[[<span class="st">"rnaseq"</span>]])</a></code></pre></div>
<pre><code>## List of length 3
## names(3): PCA UMAP TSNE</code></pre>
<div class="sourceCode" id="cb24"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb24-1" data-line-number="1"><span class="kw"><a href="https://rdrr.io/pkg/SingleCellExperiment/man/reducedDims.html">reducedDims</a></span>(fsce[[<span class="st">"rnaseq"</span>]])<span class="op">$</span>UMAP[<span class="dv">1</span><span class="op">:</span><span class="dv">4</span>, ]</a></code></pre></div>
<pre><code>##                       [,1]       [,2]
## TGCGGGTGTAGAGTGC  1.908842  0.8429150
## TGGTTCCCATCTATGG -4.010063  0.7808292
## CATATTCCACGTCAGC  2.501747 -0.7414021
## TCACAAGTCCTGCTTG  0.891331 -8.8625031</code></pre>
</div>
<div id="tidying-data-for-analysis-and-plotting" class="section level1">
<h1 class="hasAnchor">
<a href="#tidying-data-for-analysis-and-plotting" class="anchor"></a>Tidying data for analysis and plotting</h1>
<p>Several <code>tidy_</code> functions are provided to convert <code>fsce</code> into a tidy format.</p>
<div class="sourceCode" id="cb26"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb26-1" data-line-number="1"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span>(dplyr)</a></code></pre></div>
<pre><code>## 
## Attaching package: 'dplyr'</code></pre>
<pre><code>## The following object is masked from 'package:matrixStats':
## 
##     count</code></pre>
<pre><code>## The following object is masked from 'package:Biobase':
## 
##     combine</code></pre>
<pre><code>## The following objects are masked from 'package:GenomicRanges':
## 
##     intersect, setdiff, union</code></pre>
<pre><code>## The following object is masked from 'package:GenomeInfoDb':
## 
##     intersect</code></pre>
<pre><code>## The following objects are masked from 'package:IRanges':
## 
##     collapse, desc, intersect, setdiff, slice, union</code></pre>
<pre><code>## The following objects are masked from 'package:S4Vectors':
## 
##     first, intersect, rename, setdiff, setequal, union</code></pre>
<pre><code>## The following objects are masked from 'package:BiocGenerics':
## 
##     combine, intersect, setdiff, union</code></pre>
<pre><code>## The following objects are masked from 'package:stats':
## 
##     filter, lag</code></pre>
<pre><code>## The following objects are masked from 'package:base':
## 
##     intersect, setdiff, setequal, union</code></pre>
<div class="sourceCode" id="cb37"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb37-1" data-line-number="1"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span>(purrr)</a></code></pre></div>
<pre><code>## 
## Attaching package: 'purrr'</code></pre>
<pre><code>## The following object is masked from 'package:DelayedArray':
## 
##     simplify</code></pre>
<pre><code>## The following object is masked from 'package:GenomicRanges':
## 
##     reduce</code></pre>
<pre><code>## The following object is masked from 'package:IRanges':
## 
##     reduce</code></pre>
<div class="sourceCode" id="cb42"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb42-1" data-line-number="1">fsce_tidy &lt;-<span class="st"> </span>purrr<span class="op">::</span><span class="kw"><a href="https://purrr.tidyverse.org/reference/reduce.html">reduce</a></span>(</a>
<a class="sourceLine" id="cb42-2" data-line-number="2">  <span class="kw"><a href="https://rdrr.io/r/base/list.html">list</a></span>(</a>
<a class="sourceLine" id="cb42-3" data-line-number="3">    <span class="kw"><a href="../reference/tidy_dims.html">tidy_dims</a></span>(fsce_small) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb42-4" data-line-number="4"><span class="st">      </span><span class="kw"><a href="https://dplyr.tidyverse.org/reference/select.html">select</a></span>(cell_id, <span class="kw"><a href="https://dplyr.tidyverse.org/reference/reexports.html">starts_with</a></span>(<span class="st">"UMAP"</span>), <span class="op">-</span>experiment),</a>
<a class="sourceLine" id="cb42-5" data-line-number="5">    <span class="kw"><a href="../reference/tidy_coldata.html">tidy_coldata</a></span>(fsce_small),</a>
<a class="sourceLine" id="cb42-6" data-line-number="6">    <span class="kw"><a href="../reference/tidy_logcounts.html">tidy_logcounts</a></span>(fsce_small[<span class="kw"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="st">"Uracil_45"</span>, <span class="st">"riboG_44"</span>), , ]) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb42-7" data-line-number="7"><span class="st">      </span><span class="kw"><a href="https://dplyr.tidyverse.org/reference/select.html">select</a></span>(<span class="op">-</span>experiment)</a>
<a class="sourceLine" id="cb42-8" data-line-number="8">  ),</a>
<a class="sourceLine" id="cb42-9" data-line-number="9">  left_join,</a>
<a class="sourceLine" id="cb42-10" data-line-number="10">  <span class="dt">by =</span> <span class="st">"cell_id"</span></a>
<a class="sourceLine" id="cb42-11" data-line-number="11">)</a></code></pre></div>
<pre><code>## harmonizing input:
##   removing 250 sampleMap rows not in names(experiments)</code></pre>
<div class="sourceCode" id="cb44"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb44-1" data-line-number="1">fsce_tidy</a></code></pre></div>
<pre><code>## # A tibble: 250 x 8
##    cell_id  UMAP1  UMAP2 k_cluster leiden_cluster cell_cycle Uracil_45
##    &lt;chr&gt;    &lt;dbl&gt;  &lt;dbl&gt; &lt;chr&gt;     &lt;chr&gt;          &lt;chr&gt;          &lt;dbl&gt;
##  1 TGCGGG…  1.91   0.843 2         2              S              1.36 
##  2 TGGTTC… -4.01   0.781 6         4              S              2.16 
##  3 CATATT…  2.50  -0.741 2         2              G2M            1.85 
##  4 TCACAA…  0.891 -8.86  5         3              G1             0.477
##  5 GGAATA… -2.52   3.65  4         4              G1             1.30 
##  6 CTTGGC…  0.606 -8.65  5         3              G2M            1.80 
##  7 GTGAAG… -3.53   2.93  6         1              S              0.895
##  8 AGCGGT… -1.70   1.16  6         1              G1             1.80 
##  9 GGATGT…  7.48   5.61  3         6              S              0    
## 10 GTACTC… -2.30   1.47  6         1              G1             1.39 
## # … with 240 more rows, and 1 more variable: riboG_44 &lt;dbl&gt;</code></pre>
</div>
<div id="visualization" class="section level1">
<h1 class="hasAnchor">
<a href="#visualization" class="anchor"></a>Visualization</h1>
<div id="two-dimensional-embeddings" class="section level2">
<h2 class="hasAnchor">
<a href="#two-dimensional-embeddings" class="anchor"></a>Two-dimensional embeddings</h2>
<p>Features can be visulized on a two-dimensional embedding with the <code><a href="../reference/plot_dims.html">plot_dims()</a></code> function which will plot gene/function assay values.</p>
<div class="sourceCode" id="cb46"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb46-1" data-line-number="1">cowplot<span class="op">::</span><span class="kw"><a href="https://rdrr.io/pkg/cowplot/man/plot_grid.html">plot_grid</a></span>(</a>
<a class="sourceLine" id="cb46-2" data-line-number="2">  <span class="kw"><a href="../reference/plot_dims.html">plot_dims</a></span>(fsce_tidy, UMAP1, UMAP2, Uracil_<span class="dv">45</span>, <span class="dt">size =</span> <span class="dv">1</span>),</a>
<a class="sourceLine" id="cb46-3" data-line-number="3">  <span class="kw"><a href="../reference/plot_dims.html">plot_dims</a></span>(fsce_tidy, UMAP1, UMAP2, riboG_<span class="dv">44</span>, <span class="dt">size =</span> <span class="dv">1</span>)</a>
<a class="sourceLine" id="cb46-4" data-line-number="4">)</a></code></pre></div>
<p><img src="scrunchy_files/figure-html/plot_dims-1.png" width="100%"></p>
</div>
<div id="functional-activities" class="section level2">
<h2 class="hasAnchor">
<a href="#functional-activities" class="anchor"></a>Functional activities</h2>
<p>Activities can be visualized per-group with <code>plot_actvitiy()</code>. These two plots have uracil and ribonucleotide excision activities plotted per cluster (k-means, n = 6).</p>
<div class="sourceCode" id="cb47"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb47-1" data-line-number="1">cowplot<span class="op">::</span><span class="kw"><a href="https://rdrr.io/pkg/cowplot/man/plot_grid.html">plot_grid</a></span>(</a>
<a class="sourceLine" id="cb47-2" data-line-number="2">  <span class="kw"><a href="../reference/plot_activity.html">plot_activity</a></span>(fsce_tidy, Uracil_<span class="dv">45</span>, k_cluster),</a>
<a class="sourceLine" id="cb47-3" data-line-number="3">  <span class="kw"><a href="../reference/plot_activity.html">plot_activity</a></span>(fsce_tidy, riboG_<span class="dv">44</span>, k_cluster)</a>
<a class="sourceLine" id="cb47-4" data-line-number="4">)</a></code></pre></div>
<p><img src="scrunchy_files/figure-html/plot_activity-1.png" width="100%"></p>
</div>
<div id="heatmap" class="section level2">
<h2 class="hasAnchor">
<a href="#heatmap" class="anchor"></a>Heatmap</h2>
<p>These heatmaps show uracil and ribonucleotide excision activities for the U:A and riboG hairpins</p>
<div class="sourceCode" id="cb48"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb48-1" data-line-number="1">mtx &lt;-<span class="st"> </span><span class="kw"><a href="https://rdrr.io/pkg/SingleCellExperiment/man/assays.html">logcounts</a></span>(fsce_small[[<span class="st">"haircut"</span>]])</a>
<a class="sourceLine" id="cb48-2" data-line-number="2"></a>
<a class="sourceLine" id="cb48-3" data-line-number="3">rows &lt;-<span class="st"> </span><span class="kw"><a href="https://rdrr.io/r/base/paste.html">paste</a></span>(<span class="st">"Uracil"</span>, <span class="dv">1</span><span class="op">:</span><span class="dv">61</span>, <span class="dt">sep =</span> <span class="st">"_"</span>)</a>
<a class="sourceLine" id="cb48-4" data-line-number="4"><span class="kw"><a href="../reference/plot_heatmap.html">plot_heatmap</a></span>(mtx, rows, <span class="dt">name =</span> <span class="st">"Uracil"</span>)</a></code></pre></div>
<p><img src="scrunchy_files/figure-html/heatmap_uracil-1.png" width="100%"></p>
<div class="sourceCode" id="cb49"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb49-1" data-line-number="1">rows &lt;-<span class="st"> </span><span class="kw"><a href="https://rdrr.io/r/base/paste.html">paste</a></span>(<span class="st">"riboG"</span>, <span class="dv">1</span><span class="op">:</span><span class="dv">61</span>, <span class="dt">sep =</span> <span class="st">"_"</span>)</a>
<a class="sourceLine" id="cb49-2" data-line-number="2"><span class="kw"><a href="../reference/plot_heatmap.html">plot_heatmap</a></span>(mtx, rows, <span class="dt">name =</span> <span class="st">"riboG"</span>)</a></code></pre></div>
<p><img src="scrunchy_files/figure-html/heatpmap_riboG-1.png" width="100%"></p>
</div>
</div>
<div id="appendix" class="section level1">
<h1 class="hasAnchor">
<a href="#appendix" class="anchor"></a>Appendix</h1>
<div id="loading-data" class="section level3">
<h3 class="hasAnchor">
<a href="#loading-data" class="anchor"></a>Loading data</h3>
<div class="sourceCode" id="cb50"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb50-1" data-line-number="1"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span>(scrunchy)</a>
<a class="sourceLine" id="cb50-2" data-line-number="2"></a>
<a class="sourceLine" id="cb50-3" data-line-number="3"><span class="co"># this same object is available as `fsce_small` in the package.</span></a>
<a class="sourceLine" id="cb50-4" data-line-number="4">fsce &lt;-<span class="st"> </span><span class="kw"><a href="../reference/create_fsce.html">create_fsce</a></span>(</a>
<a class="sourceLine" id="cb50-5" data-line-number="5">  <span class="kw"><a href="https://rdrr.io/r/base/list.html">list</a></span>(</a>
<a class="sourceLine" id="cb50-6" data-line-number="6">    <span class="dt">rnaseq =</span> <span class="kw"><a href="../reference/create_sce_rnaseq.html">create_sce_rnaseq</a></span>(<span class="kw"><a href="../reference/scrunchy_data.html">scrunchy_data</a></span>(<span class="st">"mrna"</span>)),</a>
<a class="sourceLine" id="cb50-7" data-line-number="7">    <span class="dt">haircut =</span> <span class="kw"><a href="../reference/create_sce_haircut.html">create_sce_haircut</a></span>(<span class="kw"><a href="../reference/scrunchy_data.html">scrunchy_data</a></span>(<span class="st">"haircut"</span>))</a>
<a class="sourceLine" id="cb50-8" data-line-number="8">  )</a>
<a class="sourceLine" id="cb50-9" data-line-number="9">)</a>
<a class="sourceLine" id="cb50-10" data-line-number="10"></a>
<a class="sourceLine" id="cb50-11" data-line-number="11">fsce</a></code></pre></div>
</div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="sidebar">

        <div id="tocnav">
      <h2 class="hasAnchor">
<a href="#tocnav" class="anchor"></a>Contents</h2>
      <ul class="nav nav-pills nav-stacked">
<li><a href="#overview">Overview</a></li>
      <li><a href="#the-functional-single-cell-experiment-fsce">The Functional Single Cell Experiment (fsce)</a></li>
      <li><a href="#plots">Plots</a></li>
      <li><a href="#normalization">Normalization</a></li>
      <li><a href="#feature-selection-and-pca">Feature selection and PCA</a></li>
      <li><a href="#embed-cells-in-2d-using-umap-or-tsne">Embed cells in 2D using UMAP or tSNE</a></li>
      <li><a href="#tidying-data-for-analysis-and-plotting">Tidying data for analysis and plotting</a></li>
      <li>
<a href="#visualization">Visualization</a><ul class="nav nav-pills nav-stacked">
<li><a href="#two-dimensional-embeddings">Two-dimensional embeddings</a></li>
      <li><a href="#functional-activities">Functional activities</a></li>
      <li><a href="#heatmap">Heatmap</a></li>
      </ul>
</li>
      <li><a href="#appendix">Appendix</a></li>
      </ul>
</div>
      </div>

</div>



      <footer><div class="copyright">
  <p>Developed by <a href="https://hesselberthlab.org">Jay Hesselberth</a>, Amanda Richer, Kent Riemondy.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.4.1.</p>
</div>

      </footer>
</div>

  


  </body>
</html>
